tachyon/math/base/semigroups.h

```diff
@@ -313,17 +312,16 @@ class AdditiveSemigroup {
   // it.
   constexpr static std::vector<AddResult> GetSuccessiveScalarMuls(
       size_t size, const G& generator) {
-    std::vector<AddResult> ret;
-    ret.resize(size);
-    size_t num_elems_per_thread =
-        base::GetNumElementsPerThread(ret, kDefaultParallelThreshold);
-    OPENMP_PARALLEL_FOR(size_t i = 0; i < size; i += num_elems_per_thread) {
-      AddResult scalar_mul = generator.ScalarMul(i);
-      for (size_t j = i; j < i + num_elems_per_thread && j < size; ++j) {
-        ret[j] = scalar_mul;
-        scalar_mul += generator;
-      }
-    }
+    std::vector<AddResult> ret = base::CreateVector(size, AddResult::Zero());
```

Comment:
Can you make this change a separate commit?

Reply:
Since we are talking about removing `base::CreateVector()` which is similar to the constructor of `std::vector<>`, I just change it to `std::vector<AddResult> ret(size);`